napari/__init__.py

```diff
@@ -3,62 +3,22 @@
 except ImportError:
     __version__ = "not-installed"
 
-import os
-from distutils.version import StrictVersion
-from pathlib import Path
 
-try:
-    from qtpy import API_NAME
-except Exception as e:
-    if 'No Qt bindings could be found' in str(e):
-        raise type(e)(
-            "No Qt bindings could be found.\n\nnapari requires either PyQt5 or"
-            " PySide2 to be installed in the environment.\nTo install the "
-            'default backend (currently PyQt5), run "pip install napari[all]"'
-            '\nYou may also use "pip install napari[pyside2]" for Pyside2, '
-            'or "pip install napari[pyqt5]" for PyQt5'
-        ) from e
-    raise
-
-
-if API_NAME == 'PySide2':
-    # Set plugin path appropriately if using PySide2. This is a bug fix
-    # for when both PyQt5 and Pyside2 are installed
-    import PySide2
-
-    os.environ['QT_PLUGIN_PATH'] = str(
-        Path(PySide2.__file__).parent / 'Qt' / 'plugins'
-    )
-
-# When QT is not the specific version, we raise a warning:
-from warnings import warn
-
-from qtpy import QtCore
-
-if StrictVersion(QtCore.__version__) < StrictVersion('5.12.3'):
-    warn_message = f"""
-    napari was tested with QT library `>=5.12.3`.
-    The version installed is {QtCore.__version__}. Please report any issues with this
-    specific QT version at https://github.com/Napari/napari/issues.
-    """
-    warn(message=warn_message)
-
-import logging
-
-from vispy import app
+from .viewer import Viewer  # isort:skip
 
-# set vispy application to the appropriate qt backend
-app.use_app(API_NAME)
-del app
-# set vispy logger to show warning and errors only
-vispy_logger = logging.getLogger('vispy')
-vispy_logger.setLevel(logging.WARNING)
+# this unused import is here to fix a very strange bug.
+# there is some mysterious magical goodness in scipy stats that needs
+# to be imported early.
+# see: https://github.com/napari/napari/issues/925
+# see: https://github.com/napari/napari/issues/1347
+from scipy import stats  # noqa: F401
 
-from .viewer import Viewer  # isort:skip
 from ._qt import gui_qt
```

Comment:
Ok, I see this is the place where everything qt gets imported and you're wondering about doing the "napari.utils.event_loop" import? I'm open to that, but I'm not sure, I really understand how that will work. I might need to think on it more. Could come after this PR merges

Reply:
Actually doesn't all the Qt stuff get imported when the Viewer gets imported above during the Window import?

Reply:
With the current PR, all the Qt stuff gets imported when a Viewer is _instantiated_ (not just when it's imported) because  `from ._qt import Window` is in the init.  I'm happy to move that to a new file, but I think keeping some `get_window_class()` helper inside the init, thereby keeping the whole Qt chain import lazy is maybe a nice thing to do?

Reply:
ah right right, i get it now. Yes, let's keep Qt chain import lazy. A helper function sounds nice.